compiler/resolution/functionResolution.cpp

```diff
@@ -8768,8 +8768,7 @@ static void moveHaltForUnacceptableTypes(CallExpr* call) {
   } else if (rhsType == dtNil) {
     bool lhsIsPointer = isClassLikeOrPtr(lhsType) ||
                         lhsType == dtCVoidPtr ||
-                        lhsType == dtCFnPtr ||
-                        lhsType == dtFile;
+                        lhsType == dtCFnPtr;
```

Comment:
So this is interesting.  I suspect there's not a test of `_file` being explicitly set to `nil`, but it's something I'd expect to continue to work.  It should be a quick test to write, and if it doesn't work, I'd expect the solution to be something like:
1. add a pragma to `compiler/dyno/include/chpl/uast/PragmaList.chpl` specifically for the `c_FILE` type, similar to how `C_PTR_CLASS` has its own pragma today and attach it to `c_FILE`
2. extend `isClassLikeOrPtr` to also check that flag

Reply:
Looking into this a bit further, I'm not sure I understand the old nilability behavior of `_file`. Currently on main, this fails to compile:
```chapel
var f: _file = nil;
```
with:
```
error: illegal cast from nil to _file
```
which I believe is coming from line 4403 in functionResultion.cpp. 

Were you expecting the above to work for `_file` (and now `c_FILE`)? Or were you referring to  something else like nillable class fields with the `_file` type?

Reply:
I guess you can't directly assign `nil` to a plain `c_ptr` either (causes `error: Could not coerce 'nil' to 'c_ptr' in initialization`), but you can assign `nil` to an instantiated `c_ptr`:
```chapel
var a: [0..<5] int = 1;
var x : c_ptr = c_ptrTo(a);
x = nil;
```

This still doesn't compile though:
```chapel
extern proc fopen(name: c_string, mode: c_string): _file;
var cf: _file = fopen("./pt.py", "r");
cf = nil;
```
with:

```
error: Cannot assign to _file from nil
```

Reply:
Maybe it's specifically for the return value from a function case?

Reply:
In any case, if it didn't work on main then I think it's okay to ignore it for now, thanks for looking!

Reply:
Assigning `nil` to a `_file` via a function that returns `nil` also isn't compiling, so I'm not sure what that was for either.